ui/app/js/controllers/user_edit_controller.js

```diff
@@ -130,8 +139,12 @@ function ($scope, $modalInstance, CSRF, Permissions, user, users) {
     $scope.saving = true;
     CSRF.getToken()
     .then(function(csrf_token) {
+      if (!$scope.user.username) {
+        $scope.errors.nullValue = 'The username field cannot be empty';
+        sweetAlert("failed", $scope.errors.nullValue, "error");
+        return false;
+      }
```

Comment:
This is a good check. Make sure you change $scope.saving back to false before returning, otherwise the "Save Changes" button will not be shown again.

Reply:
Okay